src/session_impl.cpp

```diff
@@ -1585,7 +1585,8 @@ namespace aux {
 #endif
 
 				if (m_alerts.should_post<listen_failed_alert>())
-					m_alerts.emplace_alert<listen_failed_alert>(device, bind_ep, last_op
+					m_alerts.emplace_alert<listen_failed_alert>(device
+						, bind_ep.address(), bind_ep.port(), last_op
```

Comment:
technically there could still be a constructor taking an endpoint. it make simplify constructing the alert